src/server/graphql/mutations/addOrg.js

```diff
@@ -31,56 +32,58 @@ export default {
       description: 'The name of the new team'
     }
   },
-  async resolve (source, args, {authToken, dataLoader, socketId: mutatorId}) {
-    const operationId = dataLoader.share()
-    const subOptions = {mutatorId, operationId}
+  resolve: rateLimit({perMinute: 2, perHour: 8})(
```

Comment:
using custom directives is kinda a hack, especially when building a schema in JS (vs writing one with the IDL). it must sit in front of the graphql execution call. it can't do anything but throw an error. 
putting the logic inside the resolve function allows me to bin similar mutations, but that may be an anti pattern & it is slightly messier
i finally settled on using a higher order function. it's lightweight (only takes in 1 options arg) and keeps it modular.